hudi-common/src/main/java/org/apache/hudi/io/storage/rewrite/HoodieFileRewriter.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.io.storage.rewrite;
+
+import java.io.IOException;
+
+public interface HoodieFileRewriter {
```

Comment:
we can just extend the existing `HoodieFileWriter` and add a new interface method: `HoodieFileWriter#deepCopy` and only add a impl for parquet.

Reply:
Lets not overload the `FileWriter` with this feature, since its not clear if we can handle this for  all file types? FileWriter works at the record iterator level . IMO deepCopy/fastCopy like implementations work one level lower than record level iterators writers do.. 

I question if we need an interface esp in its current form where its just two methods that don't take any args. What other implementations do we want to have for the `HoodieFileRewriter`

Reply:
The interface shoud at-least cover `rewrite(inputFiles, outputFiles, cfg)` type methods?

Reply:
Maybe this interface is useful here. Since parquet and orc both can implement Binary Copy
1. Binary copy parquet at row group level 
2. Binary copy orc at stripe level.

Also as VC mentioned. Change the function to 
` long binaryCopy(List<StoragePath> inputFilePaths, List<StoragePath> outputFilePath, MessageType writeSchema, Properties props) throws IOException;
`